hammerblade/torch/tests/test_embedding.py

```diff
@@ -0,0 +1,38 @@
+"""
+Tests on torch.nn.Embedding
+04/22/2020 Lin Cheng (lc873@cornell.edu)
+"""
+
+import torch
+import torch.nn as nn
+import random
+import copy
+
+torch.manual_seed(42)
+random.seed(42)
+
+def _test_torch_nn_embedding(input, padding_idx=None):
+    if padding_idx is not None:
+        embedding = nn.Embedding(10, 3, padding_idx=padding_idx)
+    else:
+        embedding = nn.Embedding(10, 3)
+    embedding_h = copy.deepcopy(embedding).hammerblade()
+    out = embedding(input)
+    out_h = embedding_h(input.hammerblade())
+    assert out_h.device == torch.device("hammerblade")
+    assert torch.equal(out, out_h.cpu())
+
+def test_torch_nn_embedding_1():
+    input = torch.LongTensor([1, 2, 4, 5])
```

Comment:
Just curious, why do we need LongTensor here?

Reply:
Similar to what happens to sparse tensor indices -> there are internal checks on if they are int64 values. Not sure what would break if we change them to int32. For now I downcast them before offloading to HB, but we still takes in int64 in the host code